dask/array/core.py

```diff
@@ -1080,6 +1083,50 @@ def numblocks(self):
     def npartitions(self):
         return reduce(mul, self.numblocks, 1)
 
+    def compute_chunk_sizes(self):
+        """
+        Compute the chunk sizes for a Dask array. This is especially useful
+        when the chunk sizes are unknown (e.g., with conditional indexing
+        on Dask Array).
+
+        Notes
+        -----
+        This function modifies the Dask array in-place.
```

Comment:
Has there been a discussion on in-place or not? I suppose in-place makes sense, but just want to make sure..

Reply:
> Has there been a discussion on in-place or not?

Nope. I'll ping in the main thread about this.